src/contracts/yas_factory.cairo

```diff
@@ -0,0 +1,251 @@
+
+            let (pool, _) = deploy_syscall(
+                self.pool_class_hash.read(), contract_address_salt, calldata.span(), false
+            )
+                .unwrap_syscall();
+
+            self.pool.write((token_0, token_1, fee), pool);
+            // populate mapping in the reverse direction, deliberate choice to avoid the cost of comparing addresses
+            self.pool.write((token_1, token_0, fee), pool);
+
+            self.emit(PoolCreated { token_0, token_1, fee, tick_spacing, pool });
+
+            pool
+        }
+
+        fn enable_fee_amount(ref self: ContractState, fee: u32, tick_spacing: i32) {
+            self.assert_only_owner();
+            assert(fee < 1000000, 'fee cannot be gt 1000000');
+
+            let zero = IntegerTrait::<i32>::new(0, false);
+            // tick spacing is capped at 16384 to prevent the situation where tick_spacing is so large that
+            // TickBitmap#nextInitializedTickWithinOneWord overflows int24 container from a valid tick
+            // 16384 ticks represents a >5x price change with ticks of 1 bips
+            assert(
+                tick_spacing > zero && tick_spacing < IntegerTrait::<i32>::new(16384, false),
+                'wrong tick_spacing (0<ts<16384)'
+            );
+            assert(self.fee_amount_tick_spacing(fee) == zero, 'fee amount already initialized');
+
+            self.fee_amount_tick_spacing.write(fee, tick_spacing);
+            self.emit(FeeAmountEnabled { fee, tick_spacing });
+        }
+
+        fn set_owner(ref self: ContractState, new_owner: ContractAddress) {
+            self.assert_only_owner();
+            self.emit(OwnerChanged { old_owner: self.owner(), new_owner });
+            self.owner.write(new_owner);
+        }
+    }
+
+    #[generate_trait]
+    impl InternalFunctions of InternalFunctionsTrait {
+        fn assert_only_owner(self: @ContractState) {
+            assert(get_caller_address() == self.owner(), 'only owner can do this action!');
+        }
+
+        fn serialize_calldata(
+            self: @ContractState,
+            token_0: @ContractAddress,
+            token_1: @ContractAddress,
+            fee: u32,
+            tick_spacing: @i32
+        ) -> Array<felt252> {
+            let mut calldata: Array<felt252> = Default::default();
+            calldata.append(get_contract_address().into());
+            Serde::serialize(token_0, ref calldata);
+            Serde::serialize(token_1, ref calldata);
+            calldata.append(fee.into());
+            Serde::serialize(tick_spacing, ref calldata);
+
+            calldata
+        }
+
+        fn generate_salt(
+            self: @ContractState, token_0: @ContractAddress, token_1: @ContractAddress
+        ) -> felt252 {
+            let mut data = array![];
+            Serde::serialize(token_0, ref data);
+            Serde::serialize(token_1, ref data);
+            poseidon_hash_span(data.span())
+        }
```

Comment:
These functions do not read the contract's state. Move them out of the trait

Reply:
You're right!